### PR TITLE
Add safe tests for Sanity testing

### PR DESCRIPTION
**Pull request by balhar-jakub**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [ ] Feature 
- [x] Other... Please describe:
Improval of the test suite to catch potential integration issues earlier.

#### Relevant issues
https://github.com/zowe/zowe-install-packaging/issues/1292

#### Changes proposed in this PR
The API Mediation Layer squad provides further tests for nightly builds to verify integration with other Zowe components:

Retrieval of data from MVS via Gateway
Retrieval of jobs from MVS via Gateway
Authentication calls to ZOSM/f using supported methods

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR do something the person installing Zowe should know about?

No

### CODE REVIEW

File: tests/sanity/test/apiml/test-authentication-zosmf-via-gateway.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+describe('test api mediation layer zosmf authentication', function() {
+  before('verify environment variables', function() {
+    request = testUtils.verifyAndSetupEnvironment();
+    const environment = process.env;
+    username = environment.SSH_USER;
+    password = environment.SSH_PASSWD;
+  });
+
+  describe('should be able to get data from ZOSM/f ', () => {
```
Comment (jackjia-ibm) [inline tests/sanity/test/apiml/test-authentication-zosmf-via-gateway.js:33]: Think it's a typo for z/OSMF

**Comment by jackjia-ibm** [comment]
BTW, the test failures show in https://wash.zowe.org:8443/job/zowe-install-test/job/PR-1388/2/display/redirect are not caused by and related to this PR.

File: tests/sanity/test/apiml/test-authentication-zosmf-via-gateway.js (hunk 1 above)
Comment (balhar-jakub) [reply tests/sanity/test/apiml/test-authentication-zosmf-via-gateway.js]: Fixed, thanks for spotting it.

**Comment by balhar-jakub** [comment]
@jackjia-ibm Could you please merge the PR? I don't have the authorization to do so.

**Pull request merged**